File: backend/src/main/java/backend/service/UserService.java
package backend.service;

import backend.dto.request.LoginRequest;
import backend.dto.request.RegisterRequest;
import backend.dto.response.LoginResponse;
import backend.dto.response.UserResponseDTO;
import backend.model.User;
import backend.utils.ApiResponse;

import java.util.List;
import java.util.Optional;

public interface UserService {

    ApiResponse<LoginResponse> registerUser(RegisterRequest request);

    User updateUser(String userId, User updatedUser);

    void deleteUser(String userId);

    Optional<User> getUserById(String userId);

    List<User> getSuggestedUsers(String userId);

    // UserService.java
    List<String> updateUserInterests(String userId, List<String> interests);


    // ✅ New: Login method
    ApiResponse<LoginResponse> loginUser(LoginRequest loginRequest);

    List<UserResponseDTO> getAllUsers();
}

File: src/services/userService.js
import API from './api'; // ✅ You are using API instance correctly

// ✅ Register User
export const registerUser = async (userData) => {
  try {
    const response = await API.post('/users/register', userData);
    return response.data;
  } catch (error) {
    throw error.response.data;
  }
};

// ✅ Login User
export const loginUser = async (userData) => {
  try {
    const response = await API.post('/users/login', userData);
    return response.data;
  } catch (error) {
    throw error.response.data;
  }
};

// Update User Interests
export const updateUserInterests = async (userId, interests) => {
    try {
      const response = await API.put(`/users/${userId}`, { interests });
      return response.data;
    } catch (error) {
      throw error.response.data;
    }
  };
  